code/controllers/subsystem/mapping.dm

```diff
@@ -81,6 +96,101 @@ var/datum/subsystem/mapping/SSmapping
 	shuttle_templates = SSmapping.shuttle_templates
 	shelter_templates = SSmapping.shelter_templates
 
+	previous_map_config = SSmapping.previous_map_config
+	config = SSmapping.config
+	next_map_config = SSmapping.next_map_config
+
+/datum/subsystem/mapping/proc/TryLoadZ(filename, errorList, forceLevel, last)
+	var/static/dmm_suite/loader
+	if(!loader)
+		loader = new
+	if(!loader.load_map(file(filename), 0, 0, forceLevel, no_changeturf = TRUE))
+		errorList |= filename
+	if(last)
+		QDEL_NULL(loader)
+
+#define INIT_ANNOUNCE(X) world << "<span class='boldannounce'>[X]</span>"; log_world(X)
+/datum/subsystem/mapping/proc/loadWorld()
+	//if any of these fail, something has gone horribly, HORRIBLY, wrong
+	var/list/FailedZs = list()
+
+	INIT_ANNOUNCE("Loading [config.map_name]...")
+	TryLoadZ(config.GetFullMapPath(), FailedZs, 1)
+	INIT_ANNOUNCE("Loaded station!")
+
+	INIT_ANNOUNCE("Loading mining level...")
+	TryLoadZ("_maps/map_files/generic/[config.minetype].dmm", FailedZs)
+	INIT_ANNOUNCE("Loaded mining level!")
+
+	for(var/I in 1 to 5)
```

Comment:
for one, use `world.maxz` and `world.turf` for two `for(var/i in ZLEVEL_SPACEMIN to ZLEVEL_SPACEMAX)` is just a few lines above use the constants